python/tank/bootstrap/manager.py

```diff
@@ -202,6 +203,28 @@ def _set_bundle_cache_fallback_paths(self, paths):
     )
 
 
+    def _get_async_bootstrap(self):
+        """
+        Boolean property that specifies how the engine bootstrap logic should be executed.
+
+        By default, when ```async_bootstrap`` is false, the whole engine bootstrap logic
+        will be executed synchronously in the main application thread.
+
+        When ``async_bootstrap` is true, an :class:`sgtk.Sgtk` instance will be bootstrapped asynchronously
+        in a background thread, followed by launching the engine synchronously
+        in the main application thread. This will allow the main application to continue
```

Comment:
You are right, this is an artificial limitation, and it would make sense to offer the possibility of starting the engine asynchronously.

It would be straightforward to update the threading module to support both synchronous and asynchronous engine startup. When needed, a new ticket should be opened for this.